src/js/Edit/L.PM.Edit.js

```diff
@@ -35,7 +40,48 @@ const Edit = L.Class.extend({
     }else if(type === "markerPane"){
       layer.options.pane = this._map.pm.globalOptions.panes && this._map.pm.globalOptions.panes.markerPane || 'markerPane';
     }
+  },
+  _vertexValidation(type, e){
+    const marker = e.target;
+    const args = {layer: this._layer, marker, event: e };
+
+    let validationFnc = "";
+    if(type === 'move') {
+      validationFnc = "moveVertexValidation";
+    } else if(type === 'add') {
+      validationFnc = "addVertexValidation";
+    } else if(type === 'remove') {
+      validationFnc = "removeVertexValidation";
+    }
+
+    // if validation goes wrong, we return false
+    if (this.options[validationFnc] && typeof this.options[validationFnc] === "function" && !this.options[validationFnc](args)) {
+      if(type === 'move') {
+        marker._vertexResetLatLng = marker.getLatLng();
+      }
+      return false;
+    }
+
+    marker._vertexResetLatLng = null;
+    return true;
+  },
+  _vertexValidationDrag(marker){
+    // we reset the marker to the place before it was dragged. We need this, because we can't stop the drag process in a `dragstart` | `movestart` listener
+    if(marker._vertexResetLatLng){
+      marker._latlng = marker._vertexResetLatLng;
+      marker.update();
+      return false;
+    }
+    return true;
+  },
+  _vertexValidationDragEnd(marker){
+    if(marker._vertexResetLatLng){
+      marker._vertexResetLatLng = null;
+      return false;
+    }
+    return true;
```

Comment:
these validations don't seem to validate anything 🤔 
Also: where are the users validation functions being called?